Salary App/Features/SalaryForm/View/SalaryFormViewController.swift

```diff
@@ -0,0 +1,336 @@
+        return textField
+    }()
+    
+    private lazy var taxCodeStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.alignment = .fill
+        stackView.distribution = .equalSpacing
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.layer.cornerRadius = 16
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.spacing = 8
+        return stackView
+    }()
+    
+    private lazy var taxCodeLbl: UILabel = {
+        let taxCodeLbl = UILabel()
+        taxCodeLbl.textColor = .black
+        taxCodeLbl.text = "Enter Tax code: 1257L is the most common tax code"
+        taxCodeLbl.numberOfLines = 0
+        taxCodeLbl.translatesAutoresizingMaskIntoConstraints = false
+        return taxCodeLbl
+    }()
+    
+    private lazy var taxCodeTextField: UITextField = {
+        let textField = UITextField()
+        textField.layer.borderWidth = 1
+        textField.placeholder = "1257L"
+        textField.text = "1257L"
+        textField.textColor = .black
+        let paddingView = UIView(frame: CGRect(x: 0, y: 0, width: 10, height: 50))
+        textField.leftView = paddingView
+        textField.leftViewMode = .always
+        textField.keyboardType = .asciiCapableNumberPad
+        textField.layer.cornerRadius = 8
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        return textField
+    }()
+    
+    private lazy var calculateBtn: UIButton = {
+        let calculateBtn = UIButton()
+        calculateBtn.translatesAutoresizingMaskIntoConstraints = false
+        calculateBtn.backgroundColor = .systemBlue
+        calculateBtn.setTitle("Submit", for: .normal)
+        calculateBtn.layer.cornerRadius = 16
+        calculateBtn.addTarget(self, action: #selector(calculateSalary), for: .touchUpInside)
+        return calculateBtn
+    }()
+    
+    
+   private let validationManager = ValidationManager()
+   private let taxCalculator = TaxCalculator()
+   private let niCalculator = NICalaculator()
+   private let studentLoanCalulator = StudentLoanCalculator()
+
+     let salaryvm = SalaryFormViewModel(taxCalculator: TaxCalculator(), niCalculator: NICalaculator(), studentLoanCalculator: StudentLoanCalculator(), pensionContributionCalculator: PensionContributionCalculator())
+   private let salaryTypeTextField = UITextField()
+ 
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+    }
+    
+    @objc func dismissKeyboard() {
+        view.endEditing(true)
+    }
+}
+
+extension SalaryFormViewController {
+    // MARK: Calculation
+    
+    @objc func calculateSalary() {
+        let salaryValue = Double(salaryTextField.text ?? "") ?? 0
+        let pensionValue = Double(pensionTextField.text ?? "") ?? 0
+         do {
+            try validationManager.isValid(texts: [salaryTextField.text ?? "" , pensionTextField.text ?? ""])
```

Comment:
I wouln't be passing in an array of strings, this isn't the best approach... You should have one function called `isValid(input: FormInput)`. Then you do all of the validation in this one single function where you check each property and it throws an error all of the validation below needs be removed and instead you just do it in one function